### PR TITLE
Skeleton arms rework and cleaning up

### PR DESCRIPTION
**Pull request by Louka3000**
- Removed duplicated arms. Now we have a single pair of arms and detach them using `detachCompletely()` and reasign them to a different parent when we want to switch between controller arm-tracking and HMD arm-tracking.
This should make developing with arms much easier, as we only have 1 pair of arms to use, and can call `public boolean isTracking(Left/Right)ArmFromController()` to know how to make it behave. Probably should add those 2 booleans to the protocol to be read eventually. This is needed for stuff like the SlimeVR overlay to display the arms properly without having to add duplicate BoneTypes.
- Moved fancy yaw/roll rotation stuff to methods.
- Removed `updatePoseAffectedByConfig`, `setHmd` and `onlyFillWaistAndFeet`, as they didn't seem to have any real use.

### CODE REVIEW

File: src/main/java/dev/slimevr/vr/processor/TransformNode.java [hunk 1]
```diff
@@ -78,4 +78,15 @@ public void combineWithParentGlobalRotation(Transform parent) {
 			.multLocal(worldTransform.getTranslation())
 			.addLocal(parent.getTranslation());
 	}
+
+	public void detachCompletely() {
```
Comment (Eirenliel) [inline src/main/java/dev/slimevr/vr/processor/TransformNode.java:82]: Please rename to something like `detachWithChildren()`

**Review by Eirenliel — APPROVED** [review]

**Comment by Eirenliel** [comment]
Awesome refactoring, needs minor change.

File: src/main/java/dev/slimevr/vr/processor/TransformNode.java (hunk 1 above)
Comment (Louka3000) [reply src/main/java/dev/slimevr/vr/processor/TransformNode.java:82]: how about `detachParentAndChildren()`?

Comment (Eirenliel) [reply src/main/java/dev/slimevr/vr/processor/TransformNode.java:82]: It doesn't detaches parent, it detaches from parent though

**Review by Eirenliel — APPROVED** [review]

**Pull request merged**